### PR TITLE
Implement pay command 

### LINKED ISSUE #123: Add dollar sign for monetary values
Monetary values do not have a dollar sign prepended to it on the app interface.

### LINKED ISSUE #165: Implement pay command
When a student pays money, the tutor can reduce the amount owed and increase the amount paid automatically. This can be achieved by typing `pay INDEX AMOUNT`.

### PR DESCRIPTION
**Pull request by cadencjk**
In this PR, you can:
- Indicate that a student has paid a certain amount using `pay INDEX AMOUNT_PAID`, e.g. `pay 1 200`.
    - This will deduct the amount owed of the student
    - This will increase the amount paid of the student
- See a $ sign in each student cell.

Closes #165 
Closes #123 

**Note that the `INDEX` above refers to that in the `Schedule`.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/PayCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     */
+    public PayCommand parse(String args) throws ParseException {
+        try {
+            Index index = ParserUtil.parseIndex(args.trim().split("\\s+", 2)[0]);
+            Money moneyPaid = ParserUtil.parseMoney(args.trim().split("\\s+", 2)[1]);
+            return new PayCommand(index, moneyPaid);
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, PayCommand.MESSAGE_USAGE), pe);
+        } catch (ArrayIndexOutOfBoundsException e) {
```
Comment (Echomo-Xinyu) [inline src/main/java/seedu/address/logic/parser/PayCommandParser.java:29]: I think whether `Index` is out of range of `Array` is never checked in this block of code and hence there is no point to have such a `catch block` here?

File: src/main/java/seedu/address/model/person/Money.java [hunk 2]
```diff
@@ -41,6 +47,39 @@ public static boolean isValidMoney(Integer amount) {
...
+     *
+     * @param amountToSubtract the money to be subtracted.
+     * @return Money that consists of amount after subtraction.
+     * @throws CommandException if the final value is negative.
+     */
+    public Money subtract(Money amountToSubtract) throws CommandException {
+        Integer finalAmount = this.value - amountToSubtract.value;
+
+        if (finalAmount < 0) {
+            throw new CommandException(MESSAGE_NEGATIVE_AMOUNT);
```
Comment (Echomo-Xinyu) [inline src/main/java/seedu/address/model/person/Money.java:77]: Is this intended as an example of defensive coding? The negative number would not be able to pass the parsing in the first place. (as enforced by `Money::isValidMoney`)

**Review by Echomo-Xinyu — APPROVED** [review]
LGTM! A very well-written piece! except for the two minor issues I've spotted along the way.

File: src/main/java/seedu/address/logic/parser/PayCommandParser.java (hunk 1 above)
Comment (cadencjk) [reply src/main/java/seedu/address/logic/parser/PayCommandParser.java:29]: In order to use the current implementation of parseIndex() and parseMoney(), I have to split the user's command first using white spaces. If the user did not type the command correctly, for example `pay 2 {missing money}`, my split function will throw an out of bounds error. This is to catch that exception

Comment (Echomo-Xinyu) [reply src/main/java/seedu/address/logic/parser/PayCommandParser.java:29]: Ahh I see, make sense to me!

File: src/main/java/seedu/address/model/person/Money.java (hunk 2 above)
Comment (cadencjk) [reply src/main/java/seedu/address/model/person/Money.java:79]: > Is this intended as an example of defensive coding? The negative number would not be able to pass the parsing in the first place. (as enforced by `Money::isValidMoney`)

Good catch! Forgot about Money checking whether it is negative. Have implemented a new code.

**Pull request merged**